### ISSUE TITLE
Add support for diamond meme

### ISSUE THREAD

**Issue by HackerGal** [ibonwe/polygon-memes-generator#22]
Can you please add support to produce a diamond shaped meme. I want to specify a width and a height value as well. So that the meme produced for the diamond is according to he width and height I supply. Would also be nice to only ask for the width when I select to produce a diamond only. I don't want to supply a width to the other shapes if they do not need it.

**Comment by HackerGal** [comment]
Hey @Chad-Petersen. So in geometry,

> Diamond is another name for a Rhombus. Diamond shape is a Rhombus shape. A Rhombus four-sided shape where all sides have equal. length. A Rhombus is parallelogram.

Am I doing the rhombus shape instead? Or strictly a diamond? I think a diamond will be easier?

**Comment by Chad-Petersen** [comment]
Well technically they all the same thing in terms of euclidean geometry. As in the same math will apply to both a rhombus aka diamond and a parallelogram. It is just that when displaying a parallelogram the algorithm will be different. Its technically a diamond on its side. So I think you could just focus on creating a rhombus aka diamond for now.

According to wikipedia -I know its not the best source lol:

> The rhombus is often called a diamond, after the diamonds suit in playing cards which resembles the projection of an octahedral diamond, or a lozenge, though the former sometimes refers specifically to a rhombus with a 60° angle (see Polyiamond), and the latter sometimes refers specifically to a rhombus with a 45° angle.

So I think it would be best if we rather call this a rhombus. Since it seems like a diamond is just a special rhombus that has a certain angle. @zorbasd what do you think?

**Comment by HackerGal** [comment]
Well then, I'm going to put the rhombus on its toes so that it looks like a diamond. I think in this context we can differentiate between the two by the way they are displayed? For technicality sake..so that my diamond and your diamond looks the same? I think this is one of those situations where you need to be very specific. The last time you asked me to do an equilateral trianlge, mine looked like this,

`for (int i = 0; i < size; i++)
    {
        for (int j = 0; j < i; j++)
        {
            cout << " ";
        }
        for (int m = i; m < size; m++)
        {
            cout << character << " ";
        }
        cout << endl;
    }`

**Comment by HackerGal** [comment]
That was before you gave you explained your logic

**Comment by Chad-Petersen** [comment]
The spec for this I think should be as follows. Given a height you need to generate what would be called in layman's terms a diamond shape. 

So given that height = 7 and character = 'A'.

The output should resemble a diamond that at most 7 characters spanning the longest width and 7 characters spanning the longest height.

          A
       A A A
   A A A A A
A A A A A A A
   A A A A A
       A A A
          A
 
I hope this clears up any ambiguity in terms of the input and the expected output.

**Comment by HackerGal** [comment]
Yes, perfectly.

**Comment by Chad-Petersen** [comment]
This has been resolved in pull request #28.
